lib/rules/max-params.js

```diff
@@ -68,7 +74,7 @@ module.exports = {
 		const sourceCode = context.sourceCode;
 		const option = context.options[0];
 		let numParams = 3;
-		let countVoidThis = false;
+		let countThis = "except-void";
```

Comment:
Let's specify the defaults via [defaultOptions](https://eslint.org/docs/latest/extend/custom-rules#option-defaults) property?

Reply:
This request conflicts with https://github.com/eslint/eslint/pull/20236#discussion_r2451891341 : if `countThis` is not provided but `countVoidThis` is, the options do not match the schema.